varianteval/scripts/annotation_script.py

```diff
@@ -0,0 +1,124 @@
+                                '_'.join(comparison_labels), 'w', header=header)
+
+    # dictionary hierarchy for the labels and comparison vcfs:
+    # {label: {chrom: [records]}}
+    comparison_dicts = {}
+    for i in range(len(comparison_vcfs)):
+        comparison_dicts[comparison_labels[i]] = build_callset_dict(comparison_vcfs[i])
+
+    for rec in vcf_in_file.fetch():
+        for comp_label in comparison_labels:
+            for (comp_intrvl, comp_svtype) in comparison_dicts[comp_label][rec.chrom]:
+                # logic for checking SV type under the specified type constraint
+                type_constraint_met = (type_constraint == 'agnostic') or \
+                                      (type_constraint == 'same' and rec.info['SVTYPE'] == comp_svtype) or \
+                                      (type_constraint == 'different' and rec.info['SVTYPE'] != comp_svtype)
+                if type_constraint_met and overlapping_records((rec.start, rec.stop), comp_intrvl,
+                                                               overlap_threshold=overlap_threshold,
+                                                               bp_margin=bp_margin):
+                    comp_start, comp_stop = comp_intrvl
+                    comp_overlap = (min(rec.stop, comp_stop)) - (max(rec.start, comp_start))
+                    comp_dist_from_start = np.abs(rec.start - comp_start)
+                    if 'ANNOT' in rec.info:
+                        rec.info['ANNOT'] += "_%s-%d-%d-%d-%d-%d-%s" % (comp_label, comp_start, comp_stop,
+                                                                        comp_stop - comp_start, comp_overlap,
+                                                                        comp_dist_from_start, comp_svtype)
+                    else:
+                        rec.info['ANNOT'] = "%s-%d-%d-%d-%d-%d-%s" % (comp_label, comp_start, comp_stop,
+                                                                      comp_stop - comp_start, comp_overlap,
+                                                                      comp_dist_from_start, comp_svtype)
+                    if 'DB' not in rec.info:
+                        rec.info['DB'] = comp_label
+                    elif comp_label not in rec.info['DB']:
+                        rec.info['DB'] += ('_' + comp_label)
+        vcf_annt_file.write(rec)
+    vcf_in_file.close()
+    vcf_annt_file.close()
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='VCF annotation')
+    parser.add_argument('--output_dir', type=str, default='./', help='output directory for annotated files')
+    parser.add_argument('--source_vcf', type=str, help='input vcf that will have annotations add to (a copy of) it')
+    parser.add_argument('--comparison_vcfs', nargs='+', help='list of comparison/conflict databases')
+    parser.add_argument('--comparison_labels', nargs='+',
```

Comment:
consider taking a panel file as input of "label --> vcf" lines (instead of separate lists), like the script we have to call survivor